repository/src/main/java/com/epam/esm/dao/CertificateSpecification.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.model.Certificate;
+import com.epam.esm.model.Entity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class CertificateSpecification<T> implements Specification<T> {
+    @Autowired
+    private EntityManager entityManager;
+
+    private SearchQuery searchQuery;
+    private List<Predicate> predicates;
+
+    public CertificateSpecification(SearchQuery searchQuery) {
+        this.searchQuery = searchQuery;
+    }
+
+    @Override
+    public Predicate toPredicate(Root<T> root, CriteriaQuery<?> criteriaQuery, CriteriaBuilder criteriaBuilder) {
+        predicates = new ArrayList<>();
```

Comment:
Why did you use class field? unsafe

Reply:
I use predicate to fill it with data and then to use that data to resolve given task. Shall i initialize it in constructor?